html_parser.js

```diff
@@ -67,35 +67,61 @@ function get_nupaths(json, lines, i) {
         }
         i++;
     }
+}
 
+/**
+ * Determines whether a function contains a number.
+ * @param n  The String which could contain a number.
+ * https://stackoverflow.com/questions/5778020/check-whether-an-input-string-contains-a-number-in-javascript#28813213
```

Comment:
Should I also leave links to stackoverflow copy/pasted code?

Reply:
I've seen others do this in their production JavaScript code; I like it as it's more transparent and respects the original author, and without it, if copied code is identified, the implication is that if some of the code has been stolen without citation any of it could be. Always cite sources!